include/deal.II/base/sacado_product_type.h

```diff
@@ -1,6 +1,12 @@
 #ifndef dealii_sacado_product_type_h_deprecated
 #define dealii_sacado_product_type_h_deprecated
+
+#ifndef _MSC_VER
 #warning This file is deprecated. Use <deal.II/differentiation/ad/sacado_product_types.h> instead.
+#else
+#  pragma message( \
```

Comment:
since you can indent `#  pragma`, can you do the same with `#  warning` as well?